pkg/sql/executor.go

```diff
@@ -1332,14 +1344,11 @@ func (e *Executor) execStmtInOpenTxn(
 			}
 			qArgs[idx] = typedExpr
 		}
-		results := e.ExecutePreparedStatement(session, prepared, &parser.PlaceholderInfo{Values: qArgs, Types: prepared.SQLTypes})
-		if results.Empty {
-			return Result{}, nil
-		}
-		if len(results.ResultList) > 1 {
-			return Result{}, errWrongNumberOfPreparedStatements(len(results.ResultList))
+		results, err := e.ExecutePreparedStatement(session, prepared, &parser.PlaceholderInfo{Values: qArgs, Types: prepared.SQLTypes})
+		if err == nil && len(results.ResultList) == 1 {
```

Comment:
The check for the number of results here is weird, if err is nil and the length of the results is >0, we lose the results and they are not Close()d either.

I think this would read better as
```
if err != nil{
   return Result{}, err
}
// optional check:
if len(results.ResultList) > 1 {
   panic(...)
}
return results.ResultList[0], nil
```

Also perhaps `ExecutePreparedStatement` could return a single result instead of a list?